plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -472,7 +478,12 @@ func (c *CloudWatch) BuildMetricDatum(point telegraf.Metric) []*cloudwatch.Metri
 			unit = c.decorateMetricUnit(point.Name(), k)
 		}
 
-		for _, dimensions := range dimensionsList {
+		for index, dimensions := range dimensionsList {
+			//index == 0 means it's the original metrics, and if the metric name and dimension matches, skip creating
+			//metric datum
+			if index == 0 && c.isDropping(point.Name(), k) {
```

Comment:
Does the dimensionsList change over time, or is it set at startup like the drop origin configs? Is there a way we could remove the values from the dimensionsList upfront and avoid this if-statement?

Reply:
So if we are dropping all metrics, yes, we can avoid this if-statement and simply skip copy of the full dimension list(https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/outputs/cloudwatch/cloudwatch.go#L570) while processing roll-up. But now to support dropping specific metrics, this dimensionList has to be complete(contains the original raw dimensions), because we need to keep some metrics for the original dimension and some not. 
So to your question, this dimensionsList only change one time that is (https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/outputs/cloudwatch/cloudwatch.go#L416) and the index:0 is always original dimension.

Reply:
Stepping back a little, this for loop is inside another for loop that iterates though all metrics. So originally it means:
for each metrics(Point.Fields: [m1, m2, m3]), I want to create a datum for each dimension(dimensionsList: [[d1,d2], d1, d2]).
you will ends up with 9 metrics:
m1-[d1,d2], m2-[d1,d2], m3-[d1,d2]  <- this is original metrics   
m1-[d1], m2-[d1], m3-[d1]
m1-[d2], m2-[d2], m3-[d2]

The new code means, if we only want to drop m1 from original metrics, we still need dimensionsList equals to[ [d1,d2], d1, d2, because [d1,d2] dimension is needed for m2 and m3.